### PR TITLE
Expose client gRPC keepalive settings

### PR DESCRIPTION
**Pull request by severinson**

### CODE REVIEW

**Review by d80tb7 — APPROVED** [review]

File: pkg/client/connection.go [hunk 1]
```diff
@@ -51,21 +61,24 @@ func CreateApiConnectionWithCallOptions(
...
 	streamInterceptors := grpc.WithChainStreamInterceptor(grpc_retry.StreamClientInterceptor(retryOpts...))
 	dialOpts := append(additionalDialOptions,
 		defaultCallOptions,
-		keepAliveOptions,
 		unuaryInterceptors,
 		streamInterceptors,
 		transportCredentials(config),
 	)
+	// gRPC keepalive options.
+	if config.GrpcKeepAliveTime > 0 || config.GrpcKeepAliveTimeout > 0 {
```
Comment (d80tb7) [inline pkg/client/connection.go:73]: is this correct? What happens if `config.GrpcKeepAliveTime` = 0 and  `config.GrpcKeepAliveTime` = 1.  Should it be `&&`?

File: pkg/client/connection.go [hunk 2]
```diff
@@ -20,8 +20,18 @@ import (
 )
 
 type ApiConnectionDetails struct {
-	ArmadaUrl                   string
-	ArmadaRestUrl               string
+	ArmadaUrl     string
+	ArmadaRestUrl string
+	// After a duration of this time, if the client doesn't see any activity it
+	// pings the server to see if the transport is still alive.
+	// If set below 10s, a minimum value of 10s is used instead.
```
Comment (d80tb7) [inline pkg/client/connection.go:27]: Is the `max(time, 10s)` logic implemented?  I can't see it here

Comment (severinson) [reply pkg/client/connection.go:27]: It's implemented inside the gRPC keepalive logic.

File: pkg/client/connection.go (hunk 1 above)
Comment (severinson) [reply pkg/client/connection.go:73]: I didn't want to be opinionated here on which settings to allow. I prefer to put in known-good default values later once we've rolled out a server that supports frequent keepalive everywhere.

**Pull request merged**